openmdao/approximation_schemes/complex_step.py

```diff
@@ -102,23 +102,28 @@ def compute_approximations(self, system, jac=None, deriv_type='partial'):
             One of 'total' or 'partial', indicating if total or partial derivatives are
             being approximated.
         """
+        if len(self._exec_list) == 0:
+            return
+
         if jac is None:
             jac = system._jacobian
 
         if deriv_type == 'total':
             current_vec = system._outputs
         elif deriv_type == 'partial':
             current_vec = system._residuals
-        else:
-            raise ValueError('deriv_type must be one of "total" or "partial"')
```

Comment:
It seems like it would be more robust and wouldn't hurt to leave the ValueError in there... I know this is not normally called by the user, but it's looks like incomplete logic to not handle the case where neither the `if` nor the `elif` is true and just fall through (leaving an opening for future bugs to creep in, e.g. if someone typos the arg)...

Reply:
see above